data/data/vsphere/master/ignition.tf

```diff
@@ -0,0 +1,28 @@
+locals {
+  ignition_encoded = "data:text/plain;charset=utf-8;base64,${base64encode(var.ignition)}"
+}
+
+data "ignition_file" "hostname" {
+  count = var.instance_count
+
```

Comment:
why do we need static ignition config here? post cluster creation we should be able to create a valid master instance via API by relying only on the ignition config served by the mco (machine config operator).

Reply:
@enxebre not sure how else we would set the hostname of the control plane nodes which is required for "baremetal networking" which we are using with vSphere IPI.
https://github.com/openshift/enhancements/pull/148/files#diff-141f756d80782cc4aa106f6b3d264f2dR208

Reply:
Can we use the template's linux-costomization-options to set hostname?
https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html#linux-customization-options

Reply:
@jstuever that would certainly make it easier. Looks like it requires perl.
https://docs.vmware.com/en/VMware-vSphere/6.7/com.vmware.vsphere.vm_admin.doc/GUID-E63B6FAA-8D35-428D-B40C-744769845906.html#GUID-E63B6FAA-8D35-428D-B40C-744769845906

KB article regarding CoreOS specifically says: 
"Guest Customization of CoreOS from vCenter or vCloud Director is not supported."
https://kb.vmware.com/s/article/2109161

Not surpising the guest guide doesn't mention RHCOS:
https://www.vmware.com/resources/compatibility/pdf/VMware_GOS_Compatibility_Guide.pdf

Reply:
That is a shame... I was hoping maybe it was using DHCP trickery.

Reply:
Could the host look itself via mDNS and set its hostname that way? This certainly seems like a tricky problem to solve. I'm inclined to suggest we move forward but make sure to track this as something we must figure out

I don't know how close we are to realistically being able to replace control plane hosts today outside of this specific concern. @enxebre do you know?